src/manifest.rs

```diff
@@ -93,24 +95,50 @@ fn merge_dependencies(old_dep: &mut toml_edit::Item, new: &Dependency) {
     }
 }
 
-/// Print a message if the new dependency version is different from the old one.
-fn print_upgrade_if_necessary(
-    crate_name: &str,
-    old_dep: &toml_edit::Item,
-    new_version: &toml_edit::Item,
-) -> Result<()> {
-    let old_version = if str_or_1_len_table(old_dep) {
-        old_dep.clone()
+fn get_version(old_dep: &toml_edit::Item) -> Result<toml_edit::Item> {
+    if str_or_1_len_table(old_dep) {
+        Ok(old_dep.clone())
     } else if old_dep.is_table_like() {
         let version = old_dep["version"].clone();
         if version.is_none() {
-            return Err("Missing version field".into());
+            Err("Missing version field".into())
+        } else {
+            Ok(version)
         }
-        version
     } else {
         unreachable!("Invalid old dependency type")
+    }
+}
+
+fn old_version_compatible(dependency: &Dependency, toml_item: &toml_edit::Item) -> Result<bool> {
```

Comment:
It's not clear whether `toml_item` here is the old version or not. Maybe pass here a version or a string or at least document this.